### PR TITLE
feat(frontend): Use Zod for Network type

### PR DESCRIPTION
**Pull request by AntonioVentilii-DFINITY**
# Motivation

We change `Network` type to use Zod directly.

### CODE REVIEW

File: src/frontend/src/lib/types/network.ts [hunk 1]
```diff
@@ -16,18 +16,20 @@ const NetworkAppMetadataSchema = z.object({
 	explorerUrl: UrlSchema
 });
 
+const NetworkSchema = z.object({
+	id: NetworkIdSchema,
+	env: NetworkEnvironmentSchema,
+	name: z.string(),
+	icon: z.string().optional(),
```
Comment (peterpeterparker) [inline src/frontend/src/lib/types/network.ts:23]: Not for this PR but, should we be a bit more expressive on the type of the icons?

**Review by peterpeterparker — APPROVED** [review]
LGTM, thx

File: src/frontend/src/lib/types/network.ts (hunk 1 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/types/network.ts:23]: yes, define the type of string for icons, I will have a look

Comment (AntonioVentilii) [reply src/frontend/src/lib/types/network.ts:23]: Done with:

- PR #3875

**Pull request merged**